### PR TITLE
add Fable support to FsLexYacc.Runtime

### PR DESCRIPTION
**Pull request by dawedawe**
This allows easy usage of FsLexYacc-generated lexers/parsers with Fable.

### CODE REVIEW

File: src/FsLexYacc.Runtime/Lexing.fs [hunk 1]
```diff
@@ -400,7 +402,13 @@ type UnicodeTables(trans: uint16[] array, accept: uint16[]) =
                     // which covers all Unicode characters not covered in other
                     // ways
                     let baseForUnicodeCategories = numLowUnicodeChars + numSpecificUnicodeChars * 2
-                    let unicodeCategory = System.Globalization.CharUnicodeInfo.GetUnicodeCategory(inp)
+
+                    let unicodeCategory =
+#if FABLE_COMPILER
```
Comment (dawedawe) [inline src/FsLexYacc.Runtime/Lexing.fs:407]: I know that these can differ, see Remarks [here](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.charunicodeinfo.getunicodecategory?view=net-8.0)
But in practice, I think it's acceptable.

**Comment by nojaf** [comment]
I'm a little confused what this does.
Does it allow you to consume the generated `pars.fs` and `lex.fs`?
Or are you generating those file with Fable?
In what language did this work for you?

**Comment by dawedawe** [comment]
> Does it allow you to consume the generated `pars.fs` and `lex.fs`?

Yes, with this PR I can copy the generated Lexer/Parser to a fable project, add the new `FsLexYacc.Runtime` package to it, compile to Javascript and use the parser in the browser (or locally with bun, of course).
No need for WASM or a full fledged .NET backend just to parse sources and give feedback to the user.

Would be really, really helpful for me to have this.

**Review by nojaf — APPROVED** [review]
Yeah, I'm ok with having this, thanks 🤘!
(Would be cool to have a branch in https://github.com/dawedawe/agentleintroductiontofslexyacc showcasing this.)

**Comment by dawedawe** [comment]
Thanks!
A new release is all I need for the showcase.

**Comment by dawedawe** [comment]
> Yeah, I'm ok with having this, thanks 🤘! (Would be cool to have a branch in https://github.com/dawedawe/agentleintroductiontofslexyacc showcasing this.)

image

I added a new project to showcase in the browser: https://github.com/dawedawe/agentleintroductiontofslexyacc/tree/main/src/AfLang.Js

**Pull request merged**